### PR TITLE
RPS Events

### PR DESCRIPTION
**Pull request by nibanks**
Adds a number of new ETW events and corresponding tooling to breakdown the lifetime of a stream into different states so that we can get a better idea of the latency costs in a request/response style exchange.

### CODE REVIEW

File: src/core/stream.c [hunk 1]
```diff
@@ -37,6 +37,12 @@ QuicStreamInitialize(
         Status = QUIC_STATUS_OUT_OF_MEMORY;
         goto Exit;
     }
+
+    QuicTraceEvent(
```
Comment (nibanks) [inline src/core/stream.c:41]: If we're willing to break the ETW "contract" then we can move StreamOpen event here, and make a new StreamStarted event where StreamOpen is now.

File: src/plugins/QuicTrace.sln [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.29102.190
+# Visual Studio Version 17
+VisualStudioVersion = 17.1.31911.260
```
Comment (nibanks) [inline src/plugins/QuicTrace.sln:4]: I believe this automatically happened when I opened in VS 2022 (Preview). Should we check this in?

File: src/core/stream.c (hunk 1 above)
Comment (thhous-msft) [reply src/core/stream.c:41]: I don't think we want to break that. We have actively shipped code using the existing ETW manifest.

File: src/plugins/QuicTrace.sln (hunk 2 above)
Comment (thhous-msft) [reply src/plugins/QuicTrace.sln:4]: I think its fine. I imagine most users will be moving to 2022. Its difficult to get 2019 anymore (They hide the downloads).

**Pull request merged**